src/d/actor/d_a_btd.cpp

```diff
@@ -1,208 +1,2550 @@
+        i_this->m60DA = (u8)l_HIO.m46;
+    } else if (uVar1 == 3) {
+        cLib_addCalc2(&i_this->m60D0, 0.6f, 1.0f, 0.01f);
+    } else if (uVar1 == 4) {
+        cLib_addCalc2(&i_this->m60D0, 1.0, 1.0f, 0.2f);
+    } else if (uVar1 == 5) {
+        cLib_addCalc2(&i_this->m60D0, 1.0f, 1.0f, 0.005f);
+        uVar2 = 1;
+    } else if (uVar1 == 6) {
+        i_this->m60D0 = 0.0f;
+        uVar2 = 2;
+    } else if (uVar1 == 7) {
+        cLib_addCalc2(&i_this->m60D0, 1.0f, 1.0f, 0.03333f);
+        uVar2 = 2;
+        dKy_custom_timeset(l_HIO.m38);
+    } else if (uVar1 == 8) {
+        cLib_addCalc0(&i_this->m60D0, 1.0f, 0.06666f);
+        uVar2 = 3;
+    }
+    i_this->m60B0.mColor.r = (u8)(i_this->m60D8 * i_this->m60D4);
+    i_this->m60B0.mColor.g = (u8)(i_this->m60D9 * i_this->m60D4);
+    i_this->m60B0.mColor.b = (u8)(i_this->m60DA * i_this->m60D4);
+    i_this->m60B0.mPower = (REG0_F(8) + 5000.0f) * i_this->m60D4;
+    i_this->m60B0.mFluctuation = REG0_F(9);
+    switch (uVar2) {
+    case 0: {
+        dKy_custom_colset(0, 3, i_this->m60D0);
+        break;
+    }
+    case 1: {
+        dKy_custom_colset(0, 5, i_this->m60D0);
+        break;
+    }
+    case 2: {
+        dKy_custom_colset(6, 7, i_this->m60D0);
+        break;
+    }
+    case 3: {
+        dKy_custom_colset(0, 7, i_this->m60D0);
+        break;
+    }
+    }
+    i_this->mKankyoState = 0;
+    cLib_addCalc0(&i_this->m60D4, 1.0f, 0.025f);
 }
 
 /* 00007950-000081B8       .text daBtd_Execute__FP9btd_class */
-static BOOL daBtd_Execute(btd_class*) {
-    /* Nonmatching */
+static BOOL daBtd_Execute(btd_class* i_this) {
+    /* Nonmatching - cXyz, last for loop */
+    static f32 hand_co_pos_X[] = {630.0f, 750.0f, 700.0f, -630.0f, -750.0f, -700.0f};
+    static f32 hand_co_pos_Y[] = {0.0f, 100.0f, 0.0f, 0.0f, 100.0f, 0.0f};
+    static f32 hand_co_pos_Z[] = {1680.0f, 1600.0f, 1320.0f, 1680.0f, 1600.0f, 1320.0f};
+    static f32 hand_co_size_H[] = {250.0f, 200.0f, 200.0f, 250.0f, 200.0f, 200.0f};
+    static f32 hand_co_size_R[] = {30.0f, 170.0f, 180.0f, 30.0f, 170.0f, 180.0f};
+    u8 uVar1;
+    daPy_py_c* apdVar3;
+    short sVar4;
+    mDoExt_baseAnm* this_00;
+    mDoExt_baseAnm* this_01;
+    mDoExt_McaMorf* this_02;
+    fopAc_ac_c* pvVar6;
+    cM3dGSph* this_03;
+    cXyz* pDst;
+    cM3dGCyl* this_04;
+    J3DModel* pJVar9;
+    f32 dVar11;
+    float fVar12;
+    cXyz local_78;
+    cXyz local_6c;
+    cXyz local_60;
+
+    apdVar3 = daPy_getPlayerActorClass();
+    if (dComIfGp_getStartStageName()[0] == 'X') {
+        i_this->m6E15 = 0x32;
+    }
+    if (i_this->m6E90 != 0) {
+        dCam_getBody()->SetTypeForce("Boss01", NULL);
+    }
+    i_this->m6E88 = 1;
+    i_this->floor->getPos().y = i_this->m6E80;
+    i_this->m6E7C += REG0_F(15) + 30.0f;
+    if (i_this->m6E7C > REG0_F(18) + 2300.0f) {
+        i_this->m6E7C = REG0_F(18) + 2300.0f;
+    }
+    this_00 = (mDoExt_baseAnm*)get_btk(i_this);
+    this_01 = (mDoExt_baseAnm*)get_brk(i_this);
```

Comment:
Unnecessary casts. And these variables are not the `this` pointer, that's just some nonsense name Ghidra makes up.
```suggestion
    btk = get_btk(i_this);
    brk = get_brk(i_this);
```